__tests__/__src__/environment/TestEnvironment.ts

```diff
@@ -50,6 +52,60 @@ export class TestEnvironment extends BaseTestEnvironment {
         return result;
     }
 
+    /**
+     * Clean up your test environment.
+     * Deletes any temporary profiles that have been created
+     * @params {ITestEnvironment} testEnvironment - the test environment returned by createTestEnv
+     *
+     * @returns {Promise<void>} - promise fulfilled when cleanup is complete
+     * @throws errors if profiles fail to delete
+     * @memberof TestEnvironment
+     */
+    public static async cleanUp(testEnvironment: ITestEnvironment<ITestPropertiesSchema>) {
+        if (testEnvironment.tempProfiles != null) {
+            await TempTestProfiles.deleteProfiles(testEnvironment);
+        }
+        if (testEnvironment.pluginInstalled) {
+            const pluginDir = testEnvironment.workingDir + "/plugins";
+            require("rimraf").sync(pluginDir);
+        }
+
+        // Check if session exists before deleting resources
+        if (testEnvironment?.resources?.session) {
+            const session = testEnvironment.resources.session;
+            for (const file of testEnvironment.resources.localFiles) {
+                deleteLocalFile(file);
+            }
+            for (const file of testEnvironment.resources.files) {
+                deleteFiles(session, file);
+            }
+            for (const job of testEnvironment.resources.jobs) {
+                deleteJob(session, job);
+            }
+            for (const jobData of testEnvironment.resources.jobData) {
+                if (jobData.jobname && jobData.jobid) {
+                    deleteJobCommon(session, jobData);
+                } else {
+                    Imperative.console.info('Error: Missing jobname or jobid for jobData:', jobData);
+                }
+            }
+            for (const dataset of testEnvironment.resources.datasets) {
+                deleteDataset(session, dataset);
+            }
+            testEnvironment.resources = {
+                localFiles: [],
+                files: [],
+                jobs: [],
+                jobData: [],
+                datasets: [],
+                session: null // initializing session as null
+            };
```

Comment:
I believe that the TestEnvironment will benefit from having this initialization block in the setUp as well.
otherwise, many tests may not be able to run standalone due to the resources property being undefined 😋